src/anomalib/metrics/per_image/binclf_curve_numpy.py

```diff
@@ -0,0 +1,439 @@
+"""Binary classification curve (numpy-only implementation).
+
+A binary classification (binclf) matrix (TP, FP, FN, TN) is evaluated at multiple thresholds.
+
+The thresholds are shared by all instances/images, but their binclf are computed independently for each instance/image.
+
+author: jpcbertoldo
+"""
+
+# Original Code
+# Copyright (c) 2024 @jpcbertoldo
+# https://github.com/jpcbertoldo/aupimo
+# SPDX-License-Identifier: MIT
+#
+# Modified
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import itertools
+import logging
+from dataclasses import dataclass
+from functools import partial
+from typing import ClassVar
+
+import numpy as np
+from numpy import ndarray
+
+try:
+    import numba  # noqa: F401
+except ImportError:
+    HAS_NUMBA = False
+else:
+    HAS_NUMBA = True
+    from . import _binclf_curve_numba
+
+from . import _validate
+
+logger = logging.getLogger(__name__)
+
+# =========================================== CONSTANTS ===========================================
+
+
+@dataclass
+class BinclfAlgorithm:
```

Comment:
@ashwinvaidya17 
> What's the advantage of using this over enums?

i just wanted to avoid enums for the sake of simplicity, yet i wanted to put these contants together for consistency
should i switch to enum or is that ok?